tests/provider: Flow Log Error Different on GovCloud

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

This test fails on GovCloud:

```
TestAccAWSFlowLog_LogDestinationType_S3_Invalid
```

With this error: `Step 1/1, expected an error with pattern, no match on`

It appears that GovCloud and commercial partitions have slightly different errors.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_flow_log

### Terraform Configuration Files

```hcl
data "aws_partition" "current" {}

resource "aws_flow_log" "test" {
  log_destination      = "arn:${data.aws_partition.current.partition}:s3:::does-not-exist"
  log_destination_type = "s3"
  traffic_type         = "ALL"
  vpc_id               = aws_vpc.test.id
}
```

### Debug Output

```
TestAccAWSFlowLog_LogDestinationType_S3_Invalid: resource_aws_flow_log_test.go:241: Step 1/1, expected an error with pattern, no match on: Error running apply: 2020/10/20 23:04:48 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Error creating Flow Log for (vpc-02d4b974908c916d4), error: LogDestination: does-not-exist does not exist
        
--- FAIL: TestAccAWSFlowLog_LogDestinationType_S3_Invalid (17.70s)
```

### References

* #15738
* #15662

--- a/aws/resource_aws_flow_log_test.go
+++ b/aws/resource_aws_flow_log_test.go
@@ -245,7 +245,7 @@ func TestAccAWSFlowLog_LogDestinationType_S3_Invalid(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				Config:      testAccFlowLogConfig_LogDestinationType_S3_Invalid(rName),
-				ExpectError: regexp.MustCompile(`Access Denied for LogDestination`),
+				ExpectError: regexp.MustCompile(`(Access Denied for LogDestination|does not exist)`),
 			},
 		},
 	})
@@ -414,6 +414,8 @@ resource "aws_vpc" "test" {
 
 func testAccFlowLogConfig_LogDestinationType_CloudWatchLogs(rName string) string {
 	return testAccFlowLogConfigBase(rName) + fmt.Sprintf(`
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -425,7 +427,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [
@@ -469,8 +471,10 @@ resource "aws_flow_log" "test" {
 
 func testAccFlowLogConfig_LogDestinationType_S3_Invalid(rName string) string {
 	return testAccFlowLogConfigBase(rName) + `
+data "aws_partition" "current" {}
+
 resource "aws_flow_log" "test" {
-  log_destination      = "arn:aws:s3:::does-not-exist"
+  log_destination      = "arn:${data.aws_partition.current.partition}:s3:::does-not-exist"
   log_destination_type = "s3"
   traffic_type         = "ALL"
   vpc_id               = aws_vpc.test.id
@@ -489,6 +493,8 @@ resource "aws_subnet" "test" {
   }
 }
 
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -500,7 +506,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [
@@ -527,6 +533,8 @@ resource "aws_flow_log" "test" {
 
 func testAccFlowLogConfig_VPCID(rName string) string {
 	return testAccFlowLogConfigBase(rName) + fmt.Sprintf(`
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -538,7 +546,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [
@@ -565,6 +573,8 @@ resource "aws_flow_log" "test" {
 
 func testAccFlowLogConfig_LogFormat(rName string) string {
 	return testAccFlowLogConfigBase(rName) + fmt.Sprintf(`
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -576,7 +586,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [
@@ -609,6 +619,8 @@ resource "aws_flow_log" "test" {
 
 func testAccFlowLogConfigTags1(rName, tagKey1, tagValue1 string) string {
 	return testAccFlowLogConfigBase(rName) + fmt.Sprintf(`
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -620,7 +632,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [
@@ -651,6 +663,8 @@ resource "aws_flow_log" "test" {
 
 func testAccFlowLogConfigTags2(rName, tagKey1, tagValue1, tagKey2, tagValue2 string) string {
 	return testAccFlowLogConfigBase(rName) + fmt.Sprintf(`
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -662,7 +676,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [
@@ -694,6 +708,8 @@ resource "aws_flow_log" "test" {
 
 func testAccFlowLogConfig_MaxAggregationInterval(rName string) string {
 	return testAccFlowLogConfigBase(rName) + fmt.Sprintf(`
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
   name = %[1]q
 
@@ -705,7 +721,7 @@ resource "aws_iam_role" "test" {
       "Effect": "Allow",
       "Principal": {
         "Service": [
-          "ec2.amazonaws.com"
+          "ec2.${data.aws_partition.current.dns_suffix}"
         ]
       },
       "Action": [

